### PR TITLE
atlasci/internal/ci: migration analyzer

### PR DESCRIPTION
**Pull request by omarhachach**

### CODE REVIEW

File: sql/sqlcheck/sqlcheck.go [hunk 1]
```diff
@@ -133,6 +134,40 @@ var Destructive = &driverAware{
...
+			return err
+		}
+		diags = append(diags, Diagnostic{
+			Pos:  0,
+			Text: fmt.Sprintf("There was a checksum mismatch"),
+		})
+	}
+	if len(diags) > 0 {
+		p.Reporter.WriteReport(Report{
+			Text:        fmt.Sprintf("Migrations directory %q could not be verified", m.dir),
```
Comment (masseelch) [inline sql/sqlcheck/sqlcheck.go:164]:
```suggestion
			Text:        fmt.Sprintf("Migration directory %q could not be verified", m.dir),
```

Comment (masseelch) [reply sql/sqlcheck/sqlcheck.go:164]: Also we might wanna rephrase to something like "Migration directory is invalid". WDYT? @a8m

**Review by masseelch — COMMENTED** [review]
Looks good, please see comments :blush:

File: sql/sqlcheck/sqlcheck.go (hunk 1 above)
Comment (omarhachach) [reply sql/sqlcheck/sqlcheck.go:164]: I thought about if it were possible to check which line/file was changed

File: cmd/atlasci/internal/ci/run.go [hunk 2]
```diff
@@ -156,3 +157,43 @@ func (f *FileReport) WriteReport(r sqlcheck.Report) {
 func (w *TemplateWriter) WriteReport(r SummaryReport) error {
 	return w.T.Execute(w.W, r)
 }
+
+// MigrationAnalyzer implements sqlcheck.Analyzer. It validates the migration dir.
+type MigrationAnalyzer struct {
```
Comment (a8m) [inline cmd/atlasci/internal/ci/run.go:162]: Let's call it `ChecksumAnalyzer`.

File: cmd/atlasci/internal/ci/run.go [hunk 3]
```diff
@@ -156,3 +157,43 @@ func (f *FileReport) WriteReport(r sqlcheck.Report) {
...
+	Dir migrate.Dir
+}
+
+// Analyze implements the sqlcheck.Analyzer interface.
+func (m *MigrationAnalyzer) Analyze(_ context.Context, p *sqlcheck.Pass) error {
```
Comment (a8m) [inline cmd/atlasci/internal/ci/run.go:167]: Let's return an error instead and not continue to the rest of the analyzers. WDYT?

**Review by a8m — APPROVED** [review]

**Pull request merged**